### PR TITLE
OSPP 2021-结项报告-开发支持 QQ 聊天软件的 Wechaty Puppet Provider 模块

### PR DESCRIPTION
**Pull request by anaivebird**
Please make sure to check the following boxes creating your pull request, thank you very much!

- [ ] Add Blog Header, including title, author, image, categories, tags, etc.  

    For example:  

    ```md
    ---
    title: Your Great Blog Title
    author: your_github_id
    categories: project
    tags:
      - python
      - padlocal
    image: /assets/2021/02-your-blog-slug/teaser.jpg
    ---

    One line bio

    Your blog post content Your blog post content
    Your blog post content Your blog post content
    Your blog post content Your blog post content
    ```

    > It will be easier to start drafting your post by using a existing post file as your template, for example, [this one](https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_posts/2021-01-14-wechaty-puppet-service.md)

- [ ] Keep all filenames lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor.md` instead of `2017-10-06-WeChat PC Impactor.md`
- [ ] Embed video/pdf from YouTube/Bilibili/local (if there's any) by using `{% include iframe.html src="URL_or_PATH" %}`. (Learn more about how to embed file/url in the post by reading this [blog post](https://wechaty.js.org/2020/08/24/add-video-to-wechaty-blog/)
- [ ] Save photo to local before publishing, to a folder named by `${MONTH}-${YOUR_BLOG_SLUG}/` under `/assets/2021/` directory.
- [ ] Select a beautiful and meaningful teaser image for your blog
- [ ] Create your contributor profile (if you are a first time contributor)

    For excample:

    > jekyll/_contributors/lijiarui.md

    ```yaml
    ---
    name: Jiarui LI (李佳芮)
    site: https://lijiarui.github.io
    bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
    avatar: /jekyll/assets/contributors/lijiarui/avatar.png
    email: rui@juzi.bot
    twitter: https://weibo.com/u/2175505900
    ---
    ```

  Here is the example file:

  - lijiarui.md: <https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_contributors/lijiarui.md>
  - It will shown like this on wechaty website: <https://wechaty.js.org/contributors/lijiarui/>

To learn more about the contribute guideline, visit: <https://wechaty.js.org/docs/contributor-program/>

### CODE REVIEW

**Review by huan — APPROVED** [review]
LGTM

Thank you very much for summarizing the Wechaty Puppet OICQ project with great video presentations!

File: jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  - ospp
+
+image: /assets/2021/10-ospp-final-term-qq-puppet/wechaty-qq-puppet-logo.webp
+tags:
+  - summer-of-wechaty
+  - summer-2021
+  - ospp
+  - ospp-2021
+  - final-term
+  - other
```
Comment (lijiarui) [inline jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md]: Why do you add an `other` tag here?

File: jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+- 中期报告修改在PR上讨论，和预览看到的实际效果有些差异，对话中导师想要的效果和我的理解可能不太一样，一次次来回比较费时间。感谢李佳芮导师耐心的指导和审核，最终完成了blog的修正。
+
+- 与[wechaty-getting-started](https://github.com/wechaty/wechaty-getting-started)项目连接，做成一键化使用的过程中，遇到不少报错。感谢李卓桓导师的帮助下，完成了ESM（ES Modules）的支持，并跑通npm publish的github action，最终成功连接
+
+- oicq库会显示额外的命令行调试信息。已经通过查阅文档去除这些信息。
+
+## 项目成果
+
+项目仓库: <https://github.com/wechaty/wechaty-puppet-oicq>
```
Comment (lijiarui) [inline jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md:100]: Please add the npm package here as well.

And add your mentors as the maintainer of this package

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md (hunk 1 above)
Comment (anaivebird) [reply jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md]: deleted

File: jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md (hunk 2 above)
Comment (anaivebird) [reply jekyll/_posts/2021-10-06-ospp-final-term-qq-puppet.md:100]: ok, and huan has been added as the maintainer of this package.

**Review by lijiarui — APPROVED** [review]
LGTM

**Pull request merged**